### PR TITLE
Store user quiz results.

### PR DESCRIPTION
**Pull request by markroseberry**

### CODE REVIEW

**Comment by markroseberry** [comment]
This is not completely done yet, but I want to start getting feedback on it.

File: pybossa/model/user.py [hunk 1]
```diff
@@ -106,3 +106,57 @@ def public_info_keys(self):
...
+        quiz = self.info.get('quiz', {})
+        self.info['quiz'] = quiz
+        project_quiz = quiz.setdefault(str(project_id), {})
+        status = project_quiz.setdefault('status', 'in_progress')
+        result = project_quiz.setdefault('result', {'right':0, 'wrong':0})
+        return project_quiz
+
+    @staticmethod
+    def update_quiz_status(quiz, project_id):
+        from pybossa.core import project_repo
```
Comment (fiorda) [inline pybossa/model/user.py]: you could pass the project to avoid having to use `project_repo.get` again.

File: pybossa/sched.py [hunk 2]
```diff
@@ -63,12 +63,34 @@ def new_task(project_id, sched, user_id=None, user_ip=None,
         Schedulers.user_pref: get_user_pref_task,
         'depth_first_all': get_depth_first_all_task}
     scheduler = sched_map.get(sched, sched_map['default'])
-    present_gold_task = not random.randint(0, 10)
 
-    return scheduler(project_id, user_id, user_ip, external_uid,
-                     offset=offset, limit=limit, orderby=orderby, desc=desc,
-                     rand_within_priority=rand_within_priority,
-                     present_gold_task=present_gold_task)
+    gold_only = False
```
Comment (fiorda) [inline pybossa/sched.py]: This logic seems more appropriate for [this function](https://github.com/bloomberg/pybossa/blob/master/pybossa/api/__init__.py#L182)

File: pybossa/api/task_run.py [hunk 3]
```diff
@@ -132,7 +132,21 @@ def _add_created_timestamp(self, taskrun, task, guard):
         guard._remove_task_stamped(task, get_user_id_or_ip())
 
     def _after_save(self, instance):
-        mark_if_complete(instance.task_id, instance.project_id)
+        project_id = instance.project_id
+        mark_if_complete(instance.task_id, project_id)
+
+        project = project_repo.get(project_id)
+        if not project.get_quiz_enabled(): return
```
Comment (fiorda) [inline pybossa/api/task_run.py]: can the return be on a new line?

Comment (dchhabda) [reply pybossa/api/task_run.py]: we can also include not user.get_quiz_in_progress(project_id) here and return

File: pybossa/model/project.py [hunk 4]
```diff
@@ -199,4 +199,16 @@ def set_project_users(self, users):
     def get_project_users(self):
         return self.info.get('project_users', [])
 
+    def get_quiz_pass_fail(self, right_count, wrong_count):
+        quiz = self.info.get('quiz')
+        if not quiz:
+            raise Exception('Quiz is not configured for this project.')
+        if right_count >= quiz.get('correct_answers_to_pass'):
```
Comment (dchhabda) [inline pybossa/model/project.py]: can "correct_answers_to_pass" be "pass_count" instead as correct answer is implicit to pass?

Comment (markroseberry) [reply pybossa/model/project.py]: `pass_count` sounds like the number of times they passed the quiz. How about I change `correct_answers_to_pass` to just `pass` and `questions_per_quiz` to just `questions`?

Comment (dchhabda) [reply pybossa/model/project.py]: looks fine to me.

**Comment by markroseberry** [comment]
I'm closing this PR and replacing it with https://github.com/bloomberg/pybossa/pull/224. It includes the changes you requested.

**Pull request closed without merging**